uxarray/formatting_html.py

```diff
@@ -19,6 +19,9 @@ def _grid_header(grid, header_name=None):
 
 
 def _grid_sections(grid, max_items_collapse=15):
+    if grid._ds is None:
+        print(f"Warning: grid._ds is None for {type(grid)}")
+        return []
```

Comment:
Can we throw a warning here using the `warnings` module and provide the user with some extra information? Something like

```Python
import warnings

warnings.warn(
    "Grid repr requested but the grid is None. "
    "This often means it was dropped by an unsupported Xarray operation.",
    UserWarning
)
```

Reply:
done. How about custom __GroupBy_ class in `core/dataset.py` which
         handles the `uxgrid` propagation?

Reply:
> done. How about custom __GroupBy_ classmethod in `core/dataset.py` which handles the `uxgrid` propagation?